### PR TITLE
notepadqq: use qt5's mkDerivation

### PR DESCRIPTION
**Pull request by xaverdh**
###### Motivation for this change
See #65399

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @rszibele

### CODE REVIEW

**Review by worldofpeace — CHANGES_REQUESTED** [review]
This is because the executable is a bash script and `wrapQtAppsHook` only wraps ELF headers.

You should use `wrapQtApp $out/bin/notepadqq` in `preFixup`.

**Comment by xaverdh** [comment]
> This is because the executable is a bash script and `wrapQtAppsHook` only wraps ELF headers.
> 
> You should use `wrapQtApp $out/bin/notepadqq` in `preFixup`.

Thanks, should have realized it was a bash script..

File: pkgs/applications/editors/notepadqq/default.nix [hunk 1]
```diff
@@ -13,7 +12,7 @@ in stdenv.mkDerivation {
   };
 
   nativeBuildInputs = [
-    pkgconfig which qttools
+    pkgconfig which qttools wrapQtAppsHook
```
Comment (worldofpeace) [inline pkgs/applications/editors/notepadqq/default.nix]:
```suggestion
    pkgconfig which qttools
```

is added automatically

**Comment by teto** [comment]
tried a nix-review and while it compiles, it seems stuck on start:
```
(ins)[nix-shell:~/.cache/nix-review/pr-66324]$ ./results/notepadqq/bin/notepadqq 
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
```
then nothing happens

**Comment by xaverdh** [comment]
> tried a nix-review and while it compiles, it seems stuck on start:
> 
> ```
> (ins)[nix-shell:~/.cache/nix-review/pr-66324]$ ./results/notepadqq/bin/notepadqq 
> libpng warning: iCCP: known incorrect sRGB profile
> libpng warning: iCCP: known incorrect sRGB profile
> ```
> 
> then nothing happens

I also get that warning on startup, it also complains about some missing icons, but then runs just fine.

File: pkgs/applications/editors/notepadqq/default.nix [hunk 2]
```diff
@@ -1,9 +1,8 @@
-{ stdenv, fetchFromGitHub, pkgconfig, which, qtbase, qtsvg, qttools, qtwebkit}:
+{ mkDerivation, lib, fetchFromGitHub, pkgconfig, which, qtbase, qtsvg, qttools, qtwebkit, wrapQtAppsHook }:
```
Comment (worldofpeace) [inline pkgs/applications/editors/notepadqq/default.nix]:
```suggestion
{ mkDerivation, lib, fetchFromGitHub, pkgconfig, which, qtbase, qtsvg, qttools, qtwebkit }:
```

**Comment by xaverdh** [comment]
@teto can you strace the binary? I ran nix-review on this exact pr and had no trouble running the binary. So I guess there is some external state involved?

**Comment by worldofpeace** [comment]
I've noticed this in other applications since switching to qt wrappers, perhaps post the error in the tracking issue.

**Comment by teto** [comment]
Here is an strace result
https://transfer.sh/614Sd/log

**Pull request merged**